src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -0,0 +1,479 @@
+        )
+
+    def _findRowContainingText(self, text):
+        for row in range(self.rowCount()):
+            item = self.item(row, 0)
+            if item and item.text() == text:
+                return row
+        return None
+
+    def _createRowItems(self, filename, curve):
+        fileItem = qt.QStandardItem(filename)
+        fileItem.setData(curve, qt.Qt.UserRole)
+        fileItem.setData(False, _DROP_HIGHLIGHT_ROLE)
+        fileItem.setEditable(False)
+        fileItem.setDropEnabled(True)
+
+        iconItem = qt.QStandardItem()
+        removeItem = qt.QStandardItem()
+        iconItem.setEditable(False)
+
+        return fileItem, iconItem, removeItem
+
+    def fileExists(self, filename):
+        for row in range(self.getXParent().rowCount()):
+            item = self.item(row, 1)
+            if item and item.text() == filename:
+                return True
+        return False
+
+    def addUrl(self, url, node="X"):
+        if url.file_path() is not None:
+            file = silx.io.open(url.file_path())
+            data = file[url.data_path()]
+            if silx.io.is_dataset(data) and data.ndim == 1:
+                if node == "X":
+                    self._xDataset = data
+
+                    for item in self._plot1D.getItems():
+                        y = item.getInfo()
+                        if y is None:
+                            continue
+                        length = min(len(self._xDataset), len(y))
+                        item.setData(self._xDataset[:length], y[:length])
+
+                    self._addXFile(url.data_path())
+                else:
+                    if data is None:
+                        return
+                    curve = self._addPlot(self._xDataset, data)
+                    self._addYFile(url.data_path(), curve)
+
+    def _addPlot(self, x, y):
+        if x is None:
+            x = numpy.arange(len(y))
+
+        legend_name = f"Curve {self._index}"
+        self._index += 1
+
+        length = min(len(x), len(y))
+        curve = self._plot1D.addCurve(x=x[:length], y=y[:length], legend=legend_name)
+        curve.setInfo(y[()])
+        return curve
+
+    def _rowAboutToBeRemoved(self, parentIndex, first, last):
+        parentItem = self.itemFromIndex(parentIndex)
+        for row in range(first, last + 1):
+            fileItem = parentItem.child(row, 1)
+            curve = fileItem.data(qt.Qt.UserRole)
+            if curve is None:
+                return
+            self._plot1D.removeItem(curve)
+
+    def _updateYCurvesWithDefaultX(self):
+        for item in self._plot1D.getItems():
+            y = item.getInfo()
+            x = numpy.arange(len(y))
+            item.setData(x, y)
+            self._plot1D.resetZoom()
+
+    def clearAll(self):
+        self._xDataset = None
+
+        self._yParent.removeRows(0, self._yParent.rowCount())
+        
+        self._addXFile("")
+        self._addYFile("")
+
+        self._plot1D.clear()
```

Comment:
maybe those lines with the line 230 (`self._xDataset = None`) can be group together (`_reset` function ?) to
avoid duplication and ease future developments.